package.json

```diff
@@ -349,6 +349,31 @@
             "type": "boolean",
             "default": true,
             "description": "Restart Tomcat Server or not when http(s) port changes"
+          },
+          "tomcat.java.customEnv": {
+            "type": "array",
+            "items": {
+              "type": "object",
+              "title": "environment setting",
+              "required": [
+                "environmentVariable",
+                "value"
+              ],
+              "properties": {
+                "environmentVariable": {
+                  "type": "string",
+                  "minLength": 1,
+                  "description": "Name of the environment variable to set."
+                },
+                "value": {
+                  "type": "string",
+                  "description": "Value of the environment variable to set."
+                }
+              }
+            },
+            "default": [],
+            "description": "Specifies an array of environment variable names and values. These environment variable values will be passed to java when tomcat starts.",
+            "scope": "resource"
```

Comment:
Scope shoud be "window" based on your implementation, as you don't pass uri of resource when reading configuration. But regardless of implementation, I prefer "resource" for this feature, in case you have multiple workspace folders open in the same window.

Reply:
If you want to support different envs per workspace folder, you should provide a URI when calling `vscode.workspace.getConfiguration`